backend/src/main/java/ai/verta/modeldb/utils/JdbiUtil.java

```diff
@@ -0,0 +1,104 @@
+import ai.verta.modeldb.common.CommonMessages;
+import ai.verta.modeldb.common.MssqlMigrationUtil;
+import ai.verta.modeldb.config.MDBConfig;
+import io.grpc.health.v1.HealthCheckResponse;
+import java.io.File;
+import java.sql.SQLException;
+import java.util.concurrent.Executor;
+import liquibase.exception.LiquibaseException;
+import liquibase.resource.FileSystemResourceAccessor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.exception.JDBCConnectionException;
+
+public class JdbiUtil extends CommonDBUtil {
+  private static final Logger LOGGER = LogManager.getLogger(JdbiUtil.class);
+  private static JdbiUtil jdbiUtil;
+  private static String liquibaseRootFilePath;
+  private static boolean isReady = false;
+  private static MDBConfig config;
+
+  private JdbiUtil() {}
+
+  public static synchronized JdbiUtil getInstance(MDBConfig mdbConfig)
+      throws InterruptedException, SQLException {
+    if (jdbiUtil == null) {
+      jdbiUtil = new JdbiUtil();
+      config = mdbConfig;
+      liquibaseRootFilePath = "\\src\\main\\resources\\liquibase\\db-changelog-master.xml";
+
+      // Ensure the monitoring database is created
+      createDBIfNotExists(mdbConfig.getDatabase().getRdbConfiguration());
+
+      // Check mdb DB is up or not
+      boolean dbConnectionStatus =
+          checkDBConnection(
+              mdbConfig.getDatabase().getRdbConfiguration(), mdbConfig.getDatabase().getTimeout());
+      if (!dbConnectionStatus) {
+        checkDBConnectionInLoop(mdbConfig.getDatabase(), true);
+      }
+
+      LOGGER.info(CommonMessages.READY_STATUS, isReady);
+    }
+    return jdbiUtil;
+  }
+
+  public static boolean ping() {
+    boolean isValid = false;
+    try (var connection = getDBConnection(config.getDatabase().getRdbConfiguration())) {
+      isValid = connection.isValid(config.getDatabase().getTimeout());
+    } catch (JDBCConnectionException | SQLException ex) {
+      LOGGER.error(
+          "JdbiUtil ping() : ModelDB DB connection not found, got error: {}", ex.getMessage());
+    }
+
+    return isValid;
+  }
+
+  public static HealthCheckResponse.ServingStatus checkReady() {
+    try {
+      if (isReady && ping()) {
+        return HealthCheckResponse.ServingStatus.SERVING;
+      }
+      return HealthCheckResponse.ServingStatus.NOT_SERVING;
+    } catch (Exception ex) {
+      LOGGER.error("Getting error on health checkReady: {}", ex.getMessage(), ex);
+      return HealthCheckResponse.ServingStatus.NOT_SERVING;
+    }
+  }
+
+  public static HealthCheckResponse.ServingStatus checkLive() {
+    return HealthCheckResponse.ServingStatus.SERVING;
+  }
+
+  public void runLiquibaseMigration()
+      throws InterruptedException, LiquibaseException, SQLException {
+    runLiquibaseMigration(
+        config.getDatabase(),
+        liquibaseRootFilePath,
+        new FileSystemResourceAccessor(new File(System.getProperty(CommonConstants.USER_DIR))));
```

Comment:
shouldn't this be passed in (the file, not the accessor), rather than just magically assuming that the USER_DIR is the right place to get this?

Reply:
Removed this file from this PR. I will take this in new PR